### PR TITLE
Add adam alhousiki

### PR DESCRIPTION
**Pull request by AdamAlhousiki**

### CODE REVIEW

**Comment by hermabr** [comment]
Please add a description of what you did:))

**Comment by AdamAlhousiki** [comment]
Ohh I forgot! Mb

Here is what i did: 

Weight tying between token embedding and LM head 

QK-RMSNorm: per-head RMSNorm on Q and K before RoPE. 

Architecture: d_model=768, 8 layers, 12 heads, d_ff=2048, context 512. ~81M params total after weight tying.

Training: batch 128, lr 3e-3 → 3e-4 cosine (200 warmup, 17.5k iters to fit the 44 min budget (17500 took about 31 minutes), weight decay 0.1, grad clip 1.0, AdamW, bf16 autocast, torch.compile.
